### PR TITLE
Use late-initialize where possible

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:
Use late-initialize code where it is applicable to replace custom-delta code with late-initialize in the generator.yaml

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: generator.yaml [hunk 1]
```diff
@@ -499,6 +532,8 @@ resources:
         from:
           operation: DescribeModelPackage
           path: ModelApprovalStatus
+        late_initialize:
```
Comment (surajkota) [inline generator.yaml:535]: why do we need late initialize for this field?

**Review by surajkota — COMMENTED** [review]
One question inline

2 fields I am doubtful about because they might fit under server-side overrides than server-side defaults. Basically need to answer, what happens if the customer specifies a part of input in these fields?

1. `AlgorithmSpecification.MetricDefinitions` in training job - e.g. is there a way to distinguish which metricDefinitions are added by Cx vs service similar to `TrainingJobDefinition.StaticHyperParameters` in HPO

2. `OfflineStoreConfig.S3StorageConfig.ResolvedOutputS3URI` in feature group. is there a pattern to this override? like what happens when a customer specifies an S3 path prefix, code in this PR won't be able to handle this since `a.spec....` is not nil

Can be a follow up PR though, dont need to block this PR to handle these cases

File: generator.yaml (hunk 1 above)
Comment (ryansteakley) [reply generator.yaml:535]: Refer to custom_delta for model package this field is not mandatory but is however set to a default value of 	`// Default is for ModelApprovalStatus to be set to pending manual approval`

File: pkg/resource/model/custom_delta.go [hunk 2]
```diff
@@ -16,6 +16,7 @@ package model
 import (
 	ackcompare "github.com/aws-controllers-k8s/runtime/pkg/compare"
 	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (A-Hilaly) [inline pkg/resource/model/custom_delta.go:19]: Please apply `goimport` here

File: pkg/resource/model/custom_delta.go [hunk 3]
```diff
@@ -33,7 +34,20 @@ func customSetDefaults(
...
 	}
 
-	if ackcompare.IsNil(a.ko.Spec.EnableNetworkIsolation) && ackcompare.IsNotNil(b.ko.Spec.EnableNetworkIsolation) {
-		a.ko.Spec.EnableNetworkIsolation = b.ko.Spec.EnableNetworkIsolation
+	// Default value of Mode is SingleModel
+	mode := aws.String("SingleModel")
+
+	if ackcompare.IsNotNil(a.ko.Spec.Containers) && ackcompare.IsNotNil(b.ko.Spec.Containers) {
+		for index := range a.ko.Spec.Containers {
+			if ackcompare.IsNil(a.ko.Spec.Containers[index].Mode) && ackcompare.IsNotNil(b.ko.Spec.Containers[index].Mode) {
```
Comment (A-Hilaly) [inline pkg/resource/model/custom_delta.go:42]: What if `len(a.ko.Spec.Containers) > len(b.ko.Spec.Containers)`?

**Review by A-Hilaly — COMMENTED** [review]
`generator.yaml` looks good! 👍 
left two comments

**Review by mbaijal — APPROVED** [review]

File: pkg/resource/model/custom_delta.go (hunk 3 above)
Comment (ryansteakley) [reply pkg/resource/model/custom_delta.go:42]: Ideally we want to just set these server side defaults regardless however for our current unit-test implementation we need to check b, ideally we wouldn't. Can be improved on further pr

File: pkg/resource/model/custom_delta.go (hunk 2 above)
Comment (ryansteakley) [reply pkg/resource/model/custom_delta.go:19]: I downloaded and ran goimport but didn't seem to change anything? What is the expected output ?

File: generator.yaml (hunk 1 above)
Comment (surajkota) [reply generator.yaml:535]: let's remove this field from status since its part of the spec already

File: pkg/resource/model/custom_delta.go (hunk 2 above)
Comment (surajkota) [reply pkg/resource/model/custom_delta.go:19]: last I tried, it brought in a bunch of dependencies to the package.

Comment (mbaijal) [reply pkg/resource/model/custom_delta.go:19]: same, but is there a package size concern or should we just go ahead and include it ?

**Pull request merged**